homeassistant/components/tado/repairs.py

```diff
@@ -0,0 +1,34 @@
+"""Repair implementations."""
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+
+from .const import (
+    CONST_OVERLAY_MANUAL,
+    CONST_OVERLAY_TADO_DEFAULT,
+    DOMAIN,
+    WATER_HEATER_FALLBACK_REPAIR,
+)
+
+
+def manage_water_heater_fallback_issue(
+    hass: HomeAssistant,
+    water_heater_entities: list,
+    integration_overlay_fallback: str | None,
+) -> None:
+    """Notify users about water heater respecting fallback setting."""
+    if (
+        integration_overlay_fallback
+        in [CONST_OVERLAY_TADO_DEFAULT, CONST_OVERLAY_MANUAL]
+        and len(water_heater_entities) > 0
```

Comment:
We don't need to check the length of the container. Just iterate the container. If it's empty the iteration won't do anything.